src/reducers/composers.js

```diff
@@ -85,33 +75,147 @@ export const updateVisDataComposed = (state, action) => {
 export const addDataToMapComposed = (state, action) => {
 
   const {datasets, options, config} = action.payload;
+  let parsedConfig = config;
+  if (config && config.config && config.version) {
+// if passed in saved config
+    parsedConfig = KeplerGlSchema.parseSavedConfig(config);
+  }
+// Update visState store
+  let mergedState = updateVisDataComposed(state, {datasets, options, config: parsedConfig && parsedConfig.visState});
+  // Update mapState store
+  mergedState = {
+    ...mergedState,
+    mapState: stateMapConfigUpdater(mergedState.mapState, {payload: {mapState: parsedConfig && parsedConfig.mapState}})
+  };
+  // Update mapStyle store
+  mergedState = {
+    ...mergedState,
+    mapStyle: styleMapConfigUpdater(mergedState.mapStyle, {payload: {mapStyle: parsedConfig && parsedConfig.mapStyle}})
+  };
+  return mergedState;
+};
+/**
+ * Remove layer data from layer data
+ * @param state
+ * @param action {datasets, options, config}
+ * @returns state
+ */
+export const removeLayerDataComposed = (state, action) => {
+  // extract mapData
+  const {payload: mapData} = action;
+  const {key: datasetKey} = mapData;
+  const {datasets, layers} = state.visState;
+  // check datasetKey matches with current state 
+  if (!datasets[datasetKey]) {
```

Comment:
can datasets be null?, do we need to verify if it is null or not

Reply:
Yes, sometimes datasets may not contain datasetKey then we don't need to proceed further.